### PR TITLE
Add gatekeeper blog post

### PR DESCRIPTION
**Pull request by blairdrummond**
Based on feedback [here](https://github.com/shipwright-io/build/pull/845#issuecomment-891169979), repackaging a `docs` PR as a blog-post on how to configure an Allowlist on source urls using a Gatekeeper constraint.

Cheers, and let me know if you have any feedback!

### CODE REVIEW

File: content/en/blog/posts/2021-08-02-adding-gatekeeper.md/index.md [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+[![Gatekeeper](opa.png)](https://github.com/securekubernetes/securekubernetes/blob/master/docs/img/opa.png)
+
+## Only allow builds of allowed sources
+
+Unless you are using a tool like [source-to-image](https://github.com/openshift/source-to-image) to build images without root, there are risks to building arbitrary Docker images within your cluster. Because of these risks, an organization may want to limit builds to trusted sources, such as specific github organizations or an internally hosted git server. This is an example of how that can be configured using Gatekeeper.
```
Comment (gabemontero) [inline content/en/blog/posts/2021-08-02-adding-gatekeeper.md/index.md:19]: In the interest of vendor neutrality, please replace `Docker images` with `Open Container Initiative container images`, like one sees with https://github.com/containers/buildah#buildah---a-tool-that-facilitates-building-open-container-initiative-oci-container-images

Comment (gabemontero) [reply content/en/blog/posts/2021-08-02-adding-gatekeeper.md/index.md:19]: Also, source to image I do not believe is the only build tool with the ability to produce images with non-root.  Or at minimum, it is an evolving space.  How about we just say "Unless you are able to build images without root privileges, ..."

**Review by gabemontero — CHANGES_REQUESTED** [review]
Just a couple of minor suggestions wrt vendor neutrality @blairdrummond 

thanks

**Comment by gabemontero** [comment]
/lgtm

/assign @sbose78 

for the approve label, or to pull in other folks to take a look before merging

**Review by adambkaplan — APPROVED** [review]
/approve

This is an awesome blog post. I love everything about it. Thank you @blairdrummond

**Pull request merged**